### DISCUSSION TITLE
How can I invalidate access token on logout operation?

### DISCUSSION THREAD

**Discussion by tmkhan** [keycloak/keycloak#9369] [Q&A]
Hello,

I am trying to logout with keyCloak logout api, but I observe that I can still use my access token after being logout. In order to check it what I did is I copied the access token from browser and then tried the same in postman. I can login easily with the access token even when it was logged out. Is there any other api from keycloak or any suggestion that can help me to achieve logout. Thanks in advance.

Thanks,
Tipu

**Comment by denapucaco** [comment]
Try token revoke endpoint - /realms/{realm-name}/protocol/openid-connect/revoke

Details are available in https://www.keycloak.org/docs/latest/server_admin/

Apart from this the only other recommended solution I found is to use shorted access token lifespan.

**Comment by REX500** [comment]
Very strange that we cannot revoke individual tokens. Is there a reason for this?

A very valid use case could be:
- user logs in
- gets a token
- user does some action
- token still has 10 mins of lifetime before it expires
- user logs out
- hacker gets his token
- hacker has 10 mins to take advantage of his token

I understand that you have "Remove all" in place but that wouldn't fix the above use case without logging everyone else out too.
